src/revisions/revisions.service.spec.ts

```diff
@@ -97,4 +97,27 @@ describe('RevisionsService', () => {
       );
     });
   });
+
+  describe('purgeRevisions', () => {
+    it('purges the revision history', async () => {
+      const note = {} as Note;
+      note.id = 'test';
+      const revisions = [];
+      const revision1 = Revision.create('a', 'a');
+      revision1.id = 1;
+      const revision2 = Revision.create('b', 'b');
+      revision2.id = 2;
+      const revision3 = Revision.create('c', 'c');
+      revision3.id = 3;
+      revisions.push(revision1, revision2, revision3);
+      note.revisions = Promise.resolve(revisions);
+      jest.spyOn(revisionRepo, 'find').mockResolvedValueOnce(revisions);
+      jest.spyOn(service, 'getLatestRevision').mockResolvedValueOnce(revision3);
+      revisionRepo.remove = jest.fn().mockImplementation(() => {
+        return Promise.resolve(note.revisions);
+      });
+      // expected to return all the purged revisions
+      expect(await service.purgeRevisions(note)).toHaveLength(2);
```

Comment:
The test should also check if the expected revision remains in the 'database'.
You will need to change the mocked `remove` function to actually change the `revisions` array like this:
```js
revisionRepo.remove = jest
        .fn()
        .mockImplementation((deleteList: Revision[]) => {
          revisions = revisions.filter((item) => !deleteList.includes(item));
          return Promise.resolve(deleteList);
        });
```